### ISSUE TITLE
Lint/UnusedBlockArgument with named parameter

### ISSUE THREAD

**Issue by Dorian** [bbatsov/rubocop#3697]
When using named parameters in a block `e.g.: { |a:, b:| ... }` the `Lint/UnusedBlockArgument` cop asks to remove the unused parameters but removing one makes the other not work anymore.

--------

## Expected behavior

Ask to omit named parameters only if none of them is used.

## Actual behavior

Ask to omit named parameters even if one of them is used.

## Steps to reproduce the problem

Here is a `test.rb` file for instance:

```ruby
[{ a: 1, b: 2 }].map { |a:, b:| puts b }
[{ a: 1, b: 2 }].map { |a:, b:| puts nil }
```

Running Rubocop gives me:

```
Inspecting 1 file
W

Offenses:

test.rb:1:25: W: Unused block argument - a. If it's necessary, use _ or _a as an argument name to indicate that it won't be used.
[{ a: 1, b: 2 }].map { |a:, b:| puts b }
                        ^
test.rb:2:25: W: Unused block argument - a. You can omit all the arguments if you don't care about them.
[{ a: 1, b: 2 }].map { |a:, b:| puts nil }
                        ^
test.rb:2:29: W: Unused block argument - b. You can omit all the arguments if you don't care about them.
[{ a: 1, b: 2 }].map { |a:, b:| puts nil }
                            ^

1 file inspected, 3 offenses detected
```

The last two warnings are fine, but for the first one, using `_` or `_a` as an argument will give `missing keyword: _ (ArgumentError)` and `missing keyword: _a (ArgumentError)` errors when running the script.

## RuboCop version

```
0.45.0 (using Parser 2.3.1.4, running on ruby 2.3.1 x86_64-darwin16)
```

**Comment by drbrain** [comment]
Still seeing this with:

```
$ rubocop -v
0.49.1
```

A correct replacement would be `**` for unused arguments

**Comment by drbrain** [comment]
I rerun the example:

```
$ cat t.rb
[{ a: 1, b: 2 }].map { |_a:, b:| puts b }
$ ruby -v t.rb
ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-darwin18]
2
$ rubocop -v
0.68.1
$ rubocop t.rb
$ rubocop t.rb
Inspecting 1 file
W

Offenses:

t.rb:1:25: W: Lint/UnusedBlockArgument: Unused block argument - a. If it's necessary, use _ or _a as an argument name to indicate that it won't be used.
[{ a: 1, b: 2 }].map { |a:, b:| puts b }
                        ^

1 file inspected, 1 offense detected
```

So I make the suggested change:

```
$ cat t.rb
[{ a: 1, b: 2 }].map { |_a:, b:| puts b }
$ rubocop t.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
$ ruby -v t.rb
ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-darwin18]
Traceback (most recent call last):
	2: from t.rb:1:in `<main>'
	1: from t.rb:1:in `map'
t.rb:1:in `block in <main>': missing keyword: _a (ArgumentError)
```

The suggested change creates an error
